src/LiveQueryClient.js

```diff
@@ -358,12 +360,15 @@ class LiveQueryClient extends EventEmitter {
       break;
     case OP_EVENTS.SUBSCRIBED:
       if (subscription) {
+        subscription.subscribed = true;
+        subscription.subscribePromise.resolve();
         subscription.emit(SUBSCRIPTION_EMMITER_TYPES.OPEN);
       }
       break;
     case OP_EVENTS.ERROR:
       if (data.requestId) {
         if (subscription) {
+          subscription.subscribePromise.resolve();
```

Comment:
```
subscription.on('error', (error) => {}) // should be used if you want error.
```
I used `subscribePromise.resolve()` only just so that subscribe() doesn't hang and always returns subscription.

Reply:
let's say I do `query.subscribe().then(()=> { ... }).catch(() => { ... });`, what will happen if the subscription fails to complete?

Reply:
The chances of that happening are unlikely. unless something like https://github.com/parse-community/parse-server/pull/5871 where the json is invalid. But I get where you are going with this.